src/OpenTelemetry.Instrumentation.Runtime/RuntimeMetrics.cs

```diff
@@ -51,18 +51,21 @@ public RuntimeMetrics(RuntimeMetricsOptions options)
 
             if (options.IsGcEnabled)
             {
-                // TODO: Almost all the ObservableGauge should be ObservableUpDownCounter (except for CPU utilization).
-                // Replace them once ObservableUpDownCounter is available.
-                this.meter.CreateObservableGauge($"{metricPrefix}gc.count", () => GetGarbageCollectionCounts(), description: "GC Count for all generations.");
+                this.meter.CreateObservableCounter($"{metricPrefix}gc.count", () => GetGarbageCollectionCounts(), description: "Number of times garbage collection has occurred since process start.");
 
 #if NETCOREAPP3_1_OR_GREATER
-                this.meter.CreateObservableCounter($"{metricPrefix}gc.allocated.bytes", () => GC.GetTotalAllocatedBytes(), "By", "Allocation Rate.");
+                this.meter.CreateObservableCounter($"{metricPrefix}gc.allocated", () => GC.GetTotalAllocatedBytes(), unit: "By", description: "Count of the bytes allocated since the process start, this does not include any native allocations.");
 #endif
 
 #if NET6_0_OR_GREATER
-                this.meter.CreateObservableGauge($"{metricPrefix}gc.committed", () => GetGarbageCollectionCommittedBytes(), "By", description: "GC Committed Bytes.");
-                this.meter.CreateObservableGauge($"{metricPrefix}gc.heapsize", () => GetGarbageCollectionHeapSizes(), "By", "Heap size for all generations.");
-                this.meter.CreateObservableGauge($"{metricPrefix}gc.fragmentation.size", GetFragmentationSizes, description: "GC fragmentation.");
+                // TODO: change to ObservableUpDownCounter
+                this.meter.CreateObservableGauge($"{metricPrefix}gc.committed", () => GetGarbageCollectionCommittedBytes(), unit: "By", description: "The committed bytes of the managed heap, as observed during the latest garbage collection. If garbage collection has not occurred yet, the value will be unavailable.");
+
+                // TODO: change to ObservableUpDownCounter
+                this.meter.CreateObservableGauge($"{metricPrefix}gc.heap_size", () => GetGarbageCollectionHeapSizes(), unit: "By", description: "The heap size (including fragmentation), as observed during the latest garbage collection. If garbage collection has not occurred yet, the value will be unavailable.");
+
+                // TODO: change to ObservableUpDownCounter
+                this.meter.CreateObservableGauge($"{metricPrefix}gc.fragmentation.size", GetFragmentationSizes, unit: "By", description: "The heap fragmentation, as observed during the latest garbage collection. If garbage collection has not occurred yet, the value will be unavailable.");
```

Comment:
Is there a reason why "heap_size" uses underscore and "fragmentation.size" uses dot? Also the "size" seems to be implied when talking about commited bytes, but it is explicit elsewhere, is there a reason for the difference or should we make them all match?

Reply:
That was my mistake, I intended to put `fragmentation_size`.

I think there are three options and here's my thinking:
* `gc.heap` (with size implied) - I don't like this approach because if we have something like "heap NUMA slices" we'll run into trouble.
* `gc.heap_size` - if later we have something like "heap NUMA slices" we will have `gc.heap_numa_slices`.
`gc.heap.size` - if later we have something like "heap NUMA slices" we will have `gc.heap.numa_slices`, which might be easier for folks to search/group (e.g. search for "gc.heap.*" and create default dashboards).

I personally would prefer `gc.heap_size` and `gc.fragmentation_size`.